code/controllers/subsystem/job.dm

```diff
@@ -590,14 +603,190 @@ SUBSYSTEM_DEF(job)
+/datum/controller/subsystem/job/proc/load_jobs_from_config()
+	var/toml_file = "[global.config.directory]/jobconfig.toml"
 
-/datum/controller/subsystem/job/proc/LoadJobs()
-	var/jobstext = file2text("[global.config.directory]/jobs.txt")
-	for(var/datum/job/job as anything in joinable_occupations)
-		var/regex/jobs = new("[job.title]=(-1|\\d+),(-1|\\d+)")
-		jobs.Find(jobstext)
-		job.total_positions = text2num(jobs.group[1])
-		job.spawn_positions = text2num(jobs.group[2])
+	if(!legacy_mode) // this flag is set during the setup of SSconfig, and all warnings were handled there.
+		var/job_config = rustg_read_toml_file(toml_file)
+
+		for(var/datum/job/occupation as anything in joinable_occupations)
+			var/job_title = occupation.title
+			var/job_key = occupation.config_tag
+			if(!job_config[job_key]) // Job isn't listed, skip it.
+				message_admins(span_notice("[job_title] (with config key [job_key]) is missing from jobconfig.toml! Using codebase defaults.")) // List both job_title and job_key in case they de-sync over time.
+				continue
+
+			// If the value is commented out, we assume that the server operate did not want to override the codebase default values, so we skip it.
+			var/default_positions = job_config[job_key]["Total Positions"]
+			var/starting_positions = job_config[job_key]["Spawn Positions"]
+			var/playtime_requirements = job_config[job_key]["Playtime Requirements"]
+			var/required_account_age = job_config[job_key]["Required Account Age"]
+
+			if(default_positions)
+				occupation.total_positions = default_positions
+			if(starting_positions)
+				occupation.spawn_positions = starting_positions
+			if(playtime_requirements)
+				occupation.exp_requirements = playtime_requirements
+			if(required_account_age)
+				occupation.minimal_player_age = required_account_age
+
+		return
+
+	else
+		var/jobstext = file("[global.config.directory]/jobs.txt")
+		if(!fexists(jobstext))
+			return
+		for(var/datum/job/occupation as anything in joinable_occupations)
+			jobstext = file2text(jobstext)
+			var/regex/parser = new("[occupation.title]=(-1|\\d+),(-1|\\d+)")
+			parser.Find(jobstext)
+			occupation.total_positions = text2num(parser.group[1])
+			occupation.spawn_positions = text2num(parser.group[2])
+
+/// Called from an admin debug verb that generates the jobconfig.toml file and then allows the end user to download it to their machine. Returns TRUE if a file is successfully generated, FALSE otherwise.
+/datum/controller/subsystem/job/proc/generate_config(mob/user)
+	var/toml_file = "[global.config.directory]/jobconfig.toml"
+	var/jobstext = "[global.config.directory]/jobs.txt"
+	var/list/file_data = list()
+	config_documentation = initial(config_documentation) // Reset to default juuuuust in case.
+
+	if(fexists(file(toml_file)))
```

Comment:
Okay, wait, so what is going to happen when a non-host admin presses this button that's easily accessible to them on the live servers?

Reply:
It checks to see if it has a TOML file in the config directory, and if we do, we pull all the values from the existing jobconfig.toml (via `regenerate_job_config()`) and create a new TOML file that we FTP to the requesting client. If existing, all of the values that are in jobconfig.toml can be accessed by any admin by spending some time poking around all of the job datums in SSjob (since that's the end location of the data we pulled from the config-side jobconfig.toml).

Reply:
Ok so an admin can't change the default config at all just by clicking buttons right?

Reply:
No.